src/os/bluestore/BlueStore.cc

```diff
@@ -2003,6 +2003,29 @@ void BlueStore::ExtentMap::update(KeyValueDB::Transaction t,
   }
 }
 
+decltype(BlueStore::Blob::id) BlueStore::ExtentMap::allocate_spanning_blob_id()
+{
+  if(spanning_blob_map.empty())
+    return 0;
+  decltype(BlueStore::Blob::id) bid = spanning_blob_map.rbegin()->first + 1;
+  // bid is valid and available.
+  if(bid >= 0 && spanning_blob_map.find(bid) == spanning_blob_map.end())
```

Comment:
I am afraid I don't think so. We can't know whether the bid from +1 is used or not.